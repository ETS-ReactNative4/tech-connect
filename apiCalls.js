export const getLocations = async () => {
    const url = 'https://tech-connect-be.herokuapp.com/api/v1/locations'
    const response = await fetch(url)
    const locationData = await response.json()
    return locationData.data.map(location => {
      return {label: location.attributes.city}
    })
  }

export const getEmployers = async () => {
  const url = 'https://tech-connect-be.herokuapp.com/api/v1/employers'
  const response = await fetch(url)
  const employerData = await response.json()
  return employerData.data.map(location => {
    return {label: location.attributes.name}
  })
}


export const getPositions = async () => {
  const url = 'https://tech-connect-be.herokuapp.com/api/v1/positions'
  const response = await fetch(url)
  const positionData = await response.json()
  return positionData.data.map(location => {
    return {label: location.attributes.job_title}
  })
}

export const getAllUsers = async (key) => {
  const url = `https://tech-connect-be.herokuapp.com/api/v1/users?api_key=${key}`
  const response = await fetch(url)
  const userData = await response.json()
  return userData.data.map(user => {
    return {...user.attributes, id: user.id}
  })
}

export const getUsersFilter = async (key, filter, name) => {
  const url = `https://tech-connect-be.herokuapp.com/api/v1/users/find?${filter}=${name}&api_key=${key}`
  const response = await fetch(url)
  const userData = await response.json()
  return userData.data.map(user => {
    return {...user.attributes, id: user.id}
  })
}

export const getUserInfo = async (id, apiKey) => {

  const url = `https://tech-connect-be.herokuapp.com/api/v1/users/${id}?api_key=${apiKey}`
  try {
    const response = await fetch(url)
    if (!response.ok) {
      const errorText = await response.json()
      throw Error(errorText.error)
    }
    
    const userInfo = await response.json()
    const cleanedUser = {id: userInfo.data.id, ...userInfo.data.attributes}
    return cleanedUser
  } catch(err) {
    console.log(err)
  }
}