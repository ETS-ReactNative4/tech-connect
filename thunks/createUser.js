import { isLoading, hasErrored, logInUser } from '../actions'

export const createUser = (user) => {

  const url = 'https://tech-connect-be.herokuapp.com/api/v1/users'
  return async (dispatch) => {
    try {
      dispatch(isLoading(true))
      const response = await fetch(url, {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json'
        },
        body: JSON.stringify(user)
      })

      if (!response.ok) {
        const errorText = await response.json()
        throw Error(errorText.error)
      }

      dispatch(isLoading(false))
      const newUser = await response.json()
      const loggedInUser = {
        email: user.email,
        id: newUser.data.id,
        api_key: newUser.data.attributes.api_key,
      }
      dispatch(logInUser(loggedInUser))
      dispatch(hasErrored(''))
    } catch (err) {
      dispatch(hasErrored(err.message))
    }
  }
}

