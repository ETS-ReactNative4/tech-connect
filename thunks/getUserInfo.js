export const getUserInfo = async (id, apiKey) => {

  const url = `https://tech-connect-be.herokuapp.com/api/v1/users/${id}?api_key=${apiKey}`
  const key = {api_key: apiKey}
  try {
    const response = await fetch(url)
    if (!response.ok) {
      const errorText = await response.json()
      throw Error(errorText.error)
    }
    
    const userInfo = await response.json()
    const cleanedUser = {id: userInfo.data.id, ...userInfo.data.attributes}
    return cleanedUser
  } catch(err) {
    console.log(err)
  }
}