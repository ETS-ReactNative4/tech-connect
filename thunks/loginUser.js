import { isLoading, hasErrored, logInUser } from '../actions'

export const loginUser = (user) => {
  const url = 'https://tech-connect-be.herokuapp.com/api/v1/login'
  return async (dispatch) => {
    try {
      dispatch(isLoading(true))
      const response = await fetch(url, {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json'
        },
        body: JSON.stringify(user)
      })

      if (!response.ok) {
        const errorText = await response.json()
        throw Error(errorText.error)
      }

      dispatch(isLoading(false))
      const currentUser = await response.json();
      const loggedInUser = {
        email: user.email,
        id: currentUser.data.id,
        api_key: currentUser.data.attributes.api_key,
      }
      const userObject = {...loggedInUser,...currentUser.data.attributes}
      dispatch(logInUser(userObject))
      dispatch(hasErrored(''))
    } catch(err) {
      dispatch(hasErrored(err.message))
    }
  }
}

