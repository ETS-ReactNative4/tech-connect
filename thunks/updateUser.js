import { isLoading, hasErrored, updateUserAction } from '../actions'

export const updateUser = (user) => {
  const url = 'https://tech-connect-be.herokuapp.com/api/v1/users'
  return async (dispatch) => {

    try {
      dispatch(isLoading(true))
      const response = await fetch(url, {
        method: 'PATCH',
        headers: {
          'Content-Type': 'application/json'
        },
        body: JSON.stringify(user)
      })

      if(!response.ok) {
        const errorText = await response.json()
        throw Error(errorText.error)
      }

      dispatch(isLoading(false))
      const updatedUser = await response.json()
      dispatch(updateUserAction(updatedUser.data.attributes))
      dispatch(hasErrored(''))
    } catch(err) {
      dispatch(hasErrored(err.message))
    }
  }
}