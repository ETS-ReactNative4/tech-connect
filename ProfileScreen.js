import React from 'react'
import { StyleSheet, Text, TextInput, View, Button, TouchableHighlight } from 'react-native'
import { Input } from 'react-native-elements'
import { updateUser } from './thunks/updateUser'
import { connect } from 'react-redux'
import ModalSelector from 'react-native-modal-selector'
import { getLocations, getPositions, getEmployers } from './apiCalls'
import Icon from 'react-native-vector-icons/Feather'


export class ProfileScreen extends React.Component {
  constructor() {
    super()
    this.state = {
      name: '',
      bio: '',
      linkedin: '',
      github: '',
      phone_number: '',
      location: 'Denver',
      employer: 'Turing',
      position: 'Unemployed',
      locations: [],
      employers: [],
      position: []
    }
  }

  async componentDidMount() {
    const locations = await getLocations()
    const positions = await getPositions()
    const employers = await getEmployers()
    this.loadInputs()
    this.setState({
      locations,
      employers,
      positions
      })
  }

  loadInputs = () => {
    if(this.props.user.hasOwnProperty('name')) {
      const { name, phone_number, location, position, employer, github, linkedin, bio } = this.props.user
      let phoneNumber = phone_number
      this.setState({
        name,
        phone_number: phoneNumber.toString(),
        github,
        linkedin,
        bio,
        location: location.city,
        position: position.job_title,
        employer: employer.name,
        error: ''
      })
    }
  }


  handleSave = async () => {
    await this.checkInputs()
    if(this.state.error === '') {
      await this.props.updateUser({...this.state, api_key: this.props.user.api_key})
      this.props.navigation.navigate('Home')
    }
  }

  checkInputs = () => {
    const inputs = ['name', 'phone_number', 'location', 'position', 'employer', 'github', 'linkedin', 'bio']
    const emptyInput = inputs.filter(input => {
      return this.state[input] === ''
    })
    if(emptyInput.length) {
      this.setState({
        error: 'You are missing a required field(s)'
      })
    } else {
      this.setState({
        error: ''
      })
    }
  }

  render() {
    const { name, phone_number, github, linkedin, bio, employer, position, location, error } = this.state
    return (
      <View style={styles.container} >
      <Text style={styles.title}>Update Profile</Text>
        <Input
          onChangeText={(text) => this.setState({name: text})} 
          containerStyle={styles.profileInputContainer} 
          leftIconContainerStyle={styles.icon} 
          inputContainerStyle={styles.profileInput} 
          placeholder="Name"
          value={name}
          leftIcon={
            <Icon
              name='user'
              size={18}
              color='#4AA9C5'
            />
          }
        />
        <Input 
          onChangeText={(text) => this.setState({phone_number: text})} 
          containerStyle={styles.profileInputContainer} 
          leftIconContainerStyle={styles.icon} 
          inputContainerStyle={styles.profileInput} 
          placeholder="Phone Number"
          value={phone_number}
          leftIcon={
            <Icon 
              name='phone'
              size={18}
              color='#4AA9C5'
            />
          }
        />
        
        <ModalSelector
          optionTextStyle={{color: '#4AA9C5', fontSize: 20}}
          data={this.state.locations}
          initValue={"Select something!"}
          cancelButtonAccessibilityLabel={'Cancel Button'}
          onChange={(option)=>{ this.setState({location: option.label})}}>
          <Input
            containerStyle={styles.profileInputContainer} 
            leftIconContainerStyle={styles.icon} 
            inputContainerStyle={styles.profileInput} 
            leftIcon={
              <Icon
                name='map-pin'
                size={18}
                color='#4AA9C5'
              />
            }
          editable={false}
          placeholder="Location"
          value={location} />
        </ModalSelector>

        <ModalSelector
          optionTextStyle={{color: '#4AA9C5', fontSize: 20}}
          data={this.state.positions}
          initValue="Select something!"
          cancelButtonAccessibilityLabel={'Cancel Button'}
          onChange={(option)=>{ this.setState({position: option.label})}}>
          <Input
            containerStyle={styles.profileInputContainer} 
            leftIconContainerStyle={styles.icon} 
            inputContainerStyle={styles.profileInput} 
            leftIcon={
              <Icon
                name='briefcase'
                size={18}
                color='#4AA9C5'
              />
            }
          editable={false}
          placeholder="Position"
          value={position} />
        </ModalSelector>

        <ModalSelector
          optionTextStyle={{color: '#4AA9C5', fontSize: 20}}
          data={this.state.employers}
          initValue="Select something!"
          cancelButtonAccessibilityLabel={'Cancel Button'}
          onChange={(option)=>{ this.setState({employer: option.label})}}>
          <Input
            containerStyle={styles.profileInputContainer} 
            leftIconContainerStyle={styles.icon} 
            inputContainerStyle={styles.profileInput} 
            leftIcon={
              <Icon
                name='home'
                size={18}
                color='#4AA9C5'
              />
            }
          editable={false}
          placeholder="Employer"
          value={employer} />
        </ModalSelector>

        <Input 
          onChangeText={(text) => this.setState({github: text})} 
          containerStyle={styles.profileInputContainer} 
          leftIconContainerStyle={styles.icon} 
          inputContainerStyle={styles.profileInput} 
          placeholder="GitHub"
          value={github}
          leftIcon={
            <Icon
              name='github'
              size={18}
              color='#4AA9C5'
            />
          }
        />
        <Input 
          onChangeText={(text) => this.setState({linkedin: text})} 
          containerStyle={styles.profileInputContainer} 
          leftIconContainerStyle={styles.icon} 
          inputContainerStyle={styles.profileInput} 
          placeholder="LinkedIn"
          value={linkedin}
          leftIcon={
            <Icon
              name='linkedin'
              size={18}
              color='#4AA9C5'
            />
          }
        />
        <Input 
          onChangeText={(text) => this.setState({bio: text})} 
          containerStyle={styles.profileInputContainer} 
          leftIconContainerStyle={styles.icon} 
          inputContainerStyle={styles.profileInput} 
          placeholder="Bio"
          value={bio}
          leftIcon={
            <Icon
              name='edit'
              size={18}
              color='#4AA9C5'
            />
          }
        />
        <Text style={ styles.error }>{ error }</Text>
        <TouchableHighlight style={styles.button}>
          <Button title="save" color='#E8FDFF' onPress={this.handleSave}/>
        </TouchableHighlight>
      </View>  
    )
  }
}

export const mapStateToProps = (state) => ({
  user: state.user,
  isLoading: state.isLoading,
  error: state.error
})

export const mapDispatchToProps = (dispatch) => ({
  updateUser: (user) => dispatch(updateUser(user)),
})

export default connect(mapStateToProps, mapDispatchToProps)(ProfileScreen)


const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center', 
    justifyContent: 'center',
  },
  title: {
    marginBottom: 20,
    fontSize: 20,
    color: '#4AA9C5'
  },
  profileInputContainer: {
    width: 300,
    height: 50,
    marginBottom: 10,
  },
  profileInput: {
    borderBottomWidth: 1.3,
    borderBottomColor: '#4AA9C5',
  },
  icon: {
    marginRight: 10,
    marginLeft: 0
  },
  button: {
    backgroundColor: '#8FC9DB',
    borderRadius: 25,
    height: 40,
    width: 140
  },
  error: {
    color: 'red',
    marginBottom: 15
  }
})