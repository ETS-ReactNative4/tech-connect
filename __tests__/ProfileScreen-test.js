import React from 'react'
import { ProfileScreen, mapStateToProps, mapDispatchToProps } from '../ProfileScreen'
import { shallow } from 'enzyme'
import { updateUser } from '../thunks/updateUser'
import { getLocations, getPositions, getEmployers } from '../apiCalls'


jest.mock('../thunks/updateUser')
jest.mock('../apiCalls')


import renderer from 'react-test-renderer'


describe('ProfileScreen', () => {
  let wrapper
  let onPressEvent
  let navigation

  beforeEach(() => {
    navigation = { navigate: jest.fn() };
    onPressEvent = jest.fn()
    wrapper = shallow(<ProfileScreen user={{api_key: 1234}} updateUser={jest.fn} navigation={navigation} />)
  })

  it('renders the snapshot', () => {
    const tree = renderer.create(
      <ProfileScreen />
      ).toJSON();
    expect(tree).toMatchSnapshot();
  });

   it('should get all locations, positions, employers on mount', async () => {
    await wrapper.instance().componentDidMount()
    expect(getLocations).toHaveBeenCalled()
    expect(getPositions).toHaveBeenCalled()
    expect(getEmployers).toHaveBeenCalled()
  })

  it('should call handleSave on press of the save button', () => {
    const spy = jest.spyOn(wrapper.instance(), 'handleSave')
    wrapper.instance().forceUpdate()
    wrapper.find('Button').simulate('press')
    expect(spy).toHaveBeenCalled();
  })

  it('should save the updated user info', () => {
    wrapper.instance().handleSave()
    expect(updateUser).toHaveBeenCalled()
  })

  it('should nagivate on save', async () => {
    await wrapper.instance().handleSave()
    expect(wrapper.instance().props.navigation.navigate).toHaveBeenCalled()
  })

  it.skip('should update the name on change of input', () => {
    wrapper.find('Input').simulate('changeText')
    wrapper.instance().forceUpdate()
    expect(wrapper.instance().state.name).toEqual('')
  })

  describe('mapStateToProps', () => {
    it('should return a user object', () => {
      const mockState = {
        user: { name: 'Kylie' },
        connections: { name: 'Bob' },
        isLoading: false,
        error: ''
      }
      const expected = { user: mockState.user, isLoading: mockState.isLoading, error: mockState.error }
      const result = mapStateToProps(mockState)
      expect(result).toEqual(expected)
    })

    describe('mapDispatchToProps', () => {
      const mockDispatch = jest.fn()
      const url = 'user.com'
      const actionToDispatch = updateUser(url)
      const result = mapDispatchToProps(mockDispatch)

      result.updateUser(url)

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })
  })
})