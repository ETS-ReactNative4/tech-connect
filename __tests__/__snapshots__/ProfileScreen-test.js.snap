// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileScreen renders the snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#4AA9C5",
        "fontSize": 20,
        "marginBottom": 20,
      }
    }
  >
    Update Profile
  </Text>
  <View
    style={
      Object {
        "height": 50,
        "marginBottom": 10,
        "paddingHorizontal": 10,
        "width": 300,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "borderBottomColor": "#4AA9C5",
          "borderBottomWidth": 1.3,
          "borderColor": "#86939e",
          "flexDirection": "row",
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "height": 40,
            "justifyContent": "center",
            "marginLeft": 0,
            "marginRight": 10,
          }
        }
      >
        <Text />
      </View>
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Name"
        style={
          Object {
            "alignSelf": "center",
            "color": "black",
            "flex": 1,
            "fontSize": 18,
            "minHeight": 40,
          }
        }
        testID="RNE__Input__text-input"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
  <View
    style={
      Object {
        "height": 50,
        "marginBottom": 10,
        "paddingHorizontal": 10,
        "width": 300,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "borderBottomColor": "#4AA9C5",
          "borderBottomWidth": 1.3,
          "borderColor": "#86939e",
          "flexDirection": "row",
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "height": 40,
            "justifyContent": "center",
            "marginLeft": 0,
            "marginRight": 10,
          }
        }
      >
        <Text />
      </View>
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Phone Number"
        style={
          Object {
            "alignSelf": "center",
            "color": "black",
            "flex": 1,
            "fontSize": 18,
            "minHeight": 40,
          }
        }
        testID="RNE__Input__text-input"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
  <View
    style={Object {}}
  >
    <Modal
      animationType="slide"
      hardwareAccelerated={false}
      onDismiss={[Function]}
      onRequestClose={[Function]}
      supportedOrientations={
        Array [
          "portrait",
          "landscape",
        ]
      }
      transparent={true}
      visible={false}
    >
      <View
        accessible={false}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "rgba(0,0,0,0.7)",
              "flex": 1,
              "justifyContent": "center",
              "padding": "5%",
            },
            Object {},
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(255,255,255,0.8)",
                "borderRadius": 5,
                "flexShrink": 1,
                "marginBottom": 8,
                "padding": 8,
              },
              Object {},
            ]
          }
        />
        <View
          style={
            Array [
              Object {
                "alignSelf": "stretch",
              },
              Object {},
            ]
          }
        >
          <View
            accessibilityLabel="Cancel Button"
            accessible={false}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "rgba(255,255,255,0.8)",
                    "borderRadius": 5,
                    "padding": 8,
                  },
                  Object {},
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#333",
                      "fontSize": 16,
                      "textAlign": "center",
                    },
                    Object {},
                  ]
                }
              >
                cancel
              </Text>
            </View>
          </View>
        </View>
      </View>
    </Modal>
    <View
      accessible={false}
      hitSlop={
        Object {
          "bottom": 0,
          "left": 0,
          "right": 0,
          "top": 0,
        }
      }
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        pointerEvents="none"
        style={Object {}}
      >
        <View
          style={
            Object {
              "height": 50,
              "marginBottom": 10,
              "paddingHorizontal": 10,
              "width": 300,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "borderBottomColor": "#4AA9C5",
                "borderBottomWidth": 1.3,
                "borderColor": "#86939e",
                "flexDirection": "row",
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                ],
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "height": 40,
                  "justifyContent": "center",
                  "marginLeft": 0,
                  "marginRight": 10,
                }
              }
            >
              <Text />
            </View>
            <TextInput
              allowFontScaling={true}
              editable={false}
              placeholder="Location"
              style={
                Object {
                  "alignSelf": "center",
                  "color": "black",
                  "flex": 1,
                  "fontSize": 18,
                  "minHeight": 40,
                }
              }
              testID="RNE__Input__text-input"
              underlineColorAndroid="transparent"
              value="Denver"
            />
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    style={Object {}}
  >
    <Modal
      animationType="slide"
      hardwareAccelerated={false}
      onDismiss={[Function]}
      onRequestClose={[Function]}
      supportedOrientations={
        Array [
          "portrait",
          "landscape",
        ]
      }
      transparent={true}
      visible={false}
    >
      <View
        accessible={false}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "rgba(0,0,0,0.7)",
              "flex": 1,
              "justifyContent": "center",
              "padding": "5%",
            },
            Object {},
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(255,255,255,0.8)",
                "borderRadius": 5,
                "flexShrink": 1,
                "marginBottom": 8,
                "padding": 8,
              },
              Object {},
            ]
          }
        />
        <View
          style={
            Array [
              Object {
                "alignSelf": "stretch",
              },
              Object {},
            ]
          }
        >
          <View
            accessibilityLabel="Cancel Button"
            accessible={false}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "rgba(255,255,255,0.8)",
                    "borderRadius": 5,
                    "padding": 8,
                  },
                  Object {},
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#333",
                      "fontSize": 16,
                      "textAlign": "center",
                    },
                    Object {},
                  ]
                }
              >
                cancel
              </Text>
            </View>
          </View>
        </View>
      </View>
    </Modal>
    <View
      accessible={false}
      hitSlop={
        Object {
          "bottom": 0,
          "left": 0,
          "right": 0,
          "top": 0,
        }
      }
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        pointerEvents="none"
        style={Object {}}
      >
        <View
          style={
            Object {
              "height": 50,
              "marginBottom": 10,
              "paddingHorizontal": 10,
              "width": 300,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "borderBottomColor": "#4AA9C5",
                "borderBottomWidth": 1.3,
                "borderColor": "#86939e",
                "flexDirection": "row",
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                ],
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "height": 40,
                  "justifyContent": "center",
                  "marginLeft": 0,
                  "marginRight": 10,
                }
              }
            >
              <Text />
            </View>
            <TextInput
              allowFontScaling={true}
              editable={false}
              placeholder="Position"
              style={
                Object {
                  "alignSelf": "center",
                  "color": "black",
                  "flex": 1,
                  "fontSize": 18,
                  "minHeight": 40,
                }
              }
              testID="RNE__Input__text-input"
              underlineColorAndroid="transparent"
              value={Array []}
            />
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    style={Object {}}
  >
    <Modal
      animationType="slide"
      hardwareAccelerated={false}
      onDismiss={[Function]}
      onRequestClose={[Function]}
      supportedOrientations={
        Array [
          "portrait",
          "landscape",
        ]
      }
      transparent={true}
      visible={false}
    >
      <View
        accessible={false}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "rgba(0,0,0,0.7)",
              "flex": 1,
              "justifyContent": "center",
              "padding": "5%",
            },
            Object {},
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(255,255,255,0.8)",
                "borderRadius": 5,
                "flexShrink": 1,
                "marginBottom": 8,
                "padding": 8,
              },
              Object {},
            ]
          }
        />
        <View
          style={
            Array [
              Object {
                "alignSelf": "stretch",
              },
              Object {},
            ]
          }
        >
          <View
            accessibilityLabel="Cancel Button"
            accessible={false}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "rgba(255,255,255,0.8)",
                    "borderRadius": 5,
                    "padding": 8,
                  },
                  Object {},
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#333",
                      "fontSize": 16,
                      "textAlign": "center",
                    },
                    Object {},
                  ]
                }
              >
                cancel
              </Text>
            </View>
          </View>
        </View>
      </View>
    </Modal>
    <View
      accessible={false}
      hitSlop={
        Object {
          "bottom": 0,
          "left": 0,
          "right": 0,
          "top": 0,
        }
      }
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        pointerEvents="none"
        style={Object {}}
      >
        <View
          style={
            Object {
              "height": 50,
              "marginBottom": 10,
              "paddingHorizontal": 10,
              "width": 300,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "borderBottomColor": "#4AA9C5",
                "borderBottomWidth": 1.3,
                "borderColor": "#86939e",
                "flexDirection": "row",
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                ],
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "height": 40,
                  "justifyContent": "center",
                  "marginLeft": 0,
                  "marginRight": 10,
                }
              }
            >
              <Text />
            </View>
            <TextInput
              allowFontScaling={true}
              editable={false}
              placeholder="Employer"
              style={
                Object {
                  "alignSelf": "center",
                  "color": "black",
                  "flex": 1,
                  "fontSize": 18,
                  "minHeight": 40,
                }
              }
              testID="RNE__Input__text-input"
              underlineColorAndroid="transparent"
              value="Turing"
            />
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "height": 50,
        "marginBottom": 10,
        "paddingHorizontal": 10,
        "width": 300,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "borderBottomColor": "#4AA9C5",
          "borderBottomWidth": 1.3,
          "borderColor": "#86939e",
          "flexDirection": "row",
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "height": 40,
            "justifyContent": "center",
            "marginLeft": 0,
            "marginRight": 10,
          }
        }
      >
        <Text />
      </View>
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="GitHub"
        style={
          Object {
            "alignSelf": "center",
            "color": "black",
            "flex": 1,
            "fontSize": 18,
            "minHeight": 40,
          }
        }
        testID="RNE__Input__text-input"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
  <View
    style={
      Object {
        "height": 50,
        "marginBottom": 10,
        "paddingHorizontal": 10,
        "width": 300,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "borderBottomColor": "#4AA9C5",
          "borderBottomWidth": 1.3,
          "borderColor": "#86939e",
          "flexDirection": "row",
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "height": 40,
            "justifyContent": "center",
            "marginLeft": 0,
            "marginRight": 10,
          }
        }
      >
        <Text />
      </View>
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="LinkedIn"
        style={
          Object {
            "alignSelf": "center",
            "color": "black",
            "flex": 1,
            "fontSize": 18,
            "minHeight": 40,
          }
        }
        testID="RNE__Input__text-input"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
  <View
    style={
      Object {
        "height": 50,
        "marginBottom": 10,
        "paddingHorizontal": 10,
        "width": 300,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "borderBottomColor": "#4AA9C5",
          "borderBottomWidth": 1.3,
          "borderColor": "#86939e",
          "flexDirection": "row",
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "height": 40,
            "justifyContent": "center",
            "marginLeft": 0,
            "marginRight": 10,
          }
        }
      >
        <Text />
      </View>
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Bio"
        style={
          Object {
            "alignSelf": "center",
            "color": "black",
            "flex": 1,
            "fontSize": 18,
            "minHeight": 40,
          }
        }
        testID="RNE__Input__text-input"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#8FC9DB",
        "borderRadius": 25,
        "height": 40,
        "width": 140,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityStates={Array []}
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
              Object {
                "color": "#E8FDFF",
              },
            ]
          }
        >
          save
        </Text>
      </View>
    </View>
  </View>
</View>
`;
